### ISSUE TITLE
tailscale netcheck: nil pointer

### ISSUE THREAD

**Issue by peter17ji** [tailscale/tailscale#13800]
### What is the issue?

```
peter22ji@RT-AX68U-BA50:/tmp/home/root# uname -a
Linux RT-AX68U-BA50 4.1.52 #2 SMP PREEMPT Fri Oct 4 15:01:08 CST 2024 aarch64 ASUSWRT-Merlin-KoolShare
peter22ji@RT-AX68U-BA50:/tmp/home/root# tailscale netcheck
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x542f4c]

goroutine 1 [running]:
tailscale.com/net/netcheck.(*Client).GetReport(0x147fd40, {0x7f6f60, 0xb6ae98}, 0x140cea0, 0x0)
        tailscale.com@v1.76.0/net/netcheck/netcheck.go:943 +0xef8
tailscale.com/cmd/tailscale/cli.runNetcheck({0x7f6f60, 0xb6ae98}, {0x1430690, 0x0, 0x0})
        tailscale.com@v1.76.0/cmd/tailscale/cli/netcheck.go:99 +0x590
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0xb58810, {0x7f6f60, 0xb6ae98})
        github.com/peterbourgon/ff/v3@v3.4.0/ffcli/command.go:153 +0x168
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0x147f920, {0x7f6f60, 0xb6ae98})
        github.com/peterbourgon/ff/v3@v3.4.0/ffcli/command.go:157 +0x10c
tailscale.com/cmd/tailscale/cli.Run({0x1484028, 0x1, 0x1})
        tailscale.com@v1.76.0/cmd/tailscale/cli/cli.go:151 +0x46c
main.main()
        tailscale.com@v1.76.0/cmd/tailscale/tailscale.go:22 +0x104
peter22ji@RT-AX68U-BA50:/tmp/home/root# which tailscale
/koolshare/bin/tailscale
peter22ji@RT-AX68U-BA50:/tmp/home/root# tailscale version
1.76.0
  tailscale commit: 6d996464ad4ce520b165c0fd3a1213876e55df5a
  other commit: d75b8cb4b03020da974de7154d6e668a36349078
  go version: go1.23.1
```

### Steps to reproduce

Should be same with #8016 .

### Are there any recent changes that introduced the issue?

Just upgraded to 1.76.0 from 1.70.0 via "tailscale upgrade" instead of opkg.

### OS

Linux

### OS version

Asuswrt Merlin 388.8 on ASUS RT-AX68U

### Tailscale version

1.76.0

### Bug report

BUG-b1936100dca0e02deb4c59153feedfda8b5b6ec6374eea42c4ed41c9b3c42cea-20241012130815Z-e10a8c52d12195fe

**Comment by qqice** [comment]
I also encountered the error after upgrading my tailscale client to 1.76.0 on Windows. 
The error output is similar with yours:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x596df6]

goroutine 1 [running]:
tailscale.com/net/netcheck.(*Client).GetReport(0xc000189720, {0x89f3e0, 0xcbf5c0}, 0xc0000ba048, 0x0)
        tailscale.com@v1.76.0/net/netcheck/netcheck.go:943 +0x10b6
tailscale.com/cmd/tailscale/cli.runNetcheck({0x89f3e0, 0xcbf5c0}, {0x779cac?, 0xc?, 0x64?})
        tailscale.com@v1.76.0/cmd/tailscale/cli/netcheck.go:99 +0x5a5
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0x779cac?, {0x89f3e0?, 0xcbf5c0?})
        github.com/peterbourgon/ff/v3@v3.4.0/ffcli/command.go:153 +0x148
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0x779cac?, {0x89f3e0?, 0xcbf5c0?})
        github.com/peterbourgon/ff/v3@v3.4.0/ffcli/command.go:157 +0xf4
tailscale.com/cmd/tailscale/cli.Run({0xc0000740d0?, 0xc00016ff40?, 0xb6b9d?})
        tailscale.com@v1.76.0/cmd/tailscale/cli/cli.go:151 +0x310
main.main()
        tailscale.com@v1.76.0/cmd/tailscale/tailscale.go:22 +0xf3
```

**Comment by nickkhyl** [comment]
Duplicate of #13780

**Comment by nickkhyl** [comment]
The fix is currently available in 1.77.12 or newer [unstable build](https://pkgs.tailscale.com/unstable/) and will be included in the 1.76.1 patch release later this week.
